src/core/services/ripple/ripple.js

```diff
@@ -117,25 +117,34 @@ function InkRippleService ($injector) {
  * Controller used by the ripple service in order to apply ripples
  * @ngInject
  */
-function InkRippleCtrl ($scope, $element, rippleOptions, $window, $timeout, $mdUtil) {
-  this.$window    = $window;
-  this.$timeout   = $timeout;
-  this.$mdUtil    = $mdUtil;
-  this.$scope     = $scope;
-  this.$element   = $element;
-  this.options    = rippleOptions;
-  this.mousedown  = false;
-  this.ripples    = [];
-  this.timeout    = null; // Stores a reference to the most-recent ripple timeout
-  this.lastRipple = null;
+function InkRippleCtrl ($scope, $element, rippleOptions, $window, $document, $timeout, $mdUtil) {
+  this.$window          = $window;
+  this.$document        = $document;
+  this.$timeout         = $timeout;
+  this.$mdUtil          = $mdUtil;
+  this.$scope           = $scope;
+  this.options          = rippleOptions;
+  this.explode          = rippleOptions.inkExplode || false;
+  this.$$element        = $element;
+  this.$element         = null;
+  this.updateContainer(); //initialize $element
+  this.mousedown        = false;
+  this.ripples          = [];
+  this.timeout          = null; // Stores a reference to the most-recent ripple timeout
+  this.lastRipple       = null;
+  this.duration         = null;
+  this.updateDuration(); //initialize duration
+
 
   $mdUtil.valueOnUse(this, 'container', this.createContainer);
 
-  this.$element.addClass('md-ink-ripple');
+  this.$$element.addClass('md-ink-ripple');
```

Comment:
Double dollar is weird for the element. I would either create a new variable, just for the explode target.